pkg/publish/ingress.go

```diff
@@ -176,74 +186,97 @@ func Ingress(req router.Request, svc *v1.ServiceInstance) (result []kclient.Obje
-	}
+		targetJSON, err := json.Marshal(targets)
+		if err != nil {
+			return nil, err
+		}
 
-	targetJSON, err := json.Marshal(targets)
-	if err != nil {
-		return nil, err
-	}
+		proto := v1.PublishProtocolHTTP
+		if len(ingressTLS) > 0 {
+			proto = v1.PublishProtocolHTTPS
+		}
 
-	proto := v1.PublishProtocolHTTP
-	if len(ingressTLS) > 0 {
-		proto = v1.PublishProtocolHTTPS
-	}
+		hostnameSeen := map[string]struct{}{}
+		for _, rule := range rules.rules {
+			if _, ok := hostnameSeen[rule.Host]; ok {
+				continue
+			}
+			hostnameSeen[rule.Host] = struct{}{}
+			svc.Status.Endpoints = append(svc.Status.Endpoints, v1.Endpoint{
+				Address:         rule.Host,
+				PublishProtocol: proto,
+			})
+		}
 
-	hostnameSeen := map[string]struct{}{}
-	for _, rule := range rules {
-		if _, ok := hostnameSeen[rule.Host]; ok {
-			continue
+		ingress := &networkingv1.Ingress{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name.SafeConcatName(svc.Name, rules.name),
+				Namespace: svc.Namespace,
+				Labels:    svc.Spec.Labels,
+				Annotations: labels.Merge(svc.Spec.Annotations, map[string]string{
+					labels.AcornTargets: string(targetJSON),
+				}),
+			},
+			Spec: networkingv1.IngressSpec{
+				IngressClassName: ingressClassName,
+				Rules:            rules.rules,
+				TLS:              ingressTLS,
+			},
+		}
+		// if custom domains don't use cert-manager, use the default acorn cert-manager setting
+		if useCertManager && (ingress.Annotations["cert-manager.io/cluster-issuer"] == "" && ingress.Annotations["cert-manager.io/issuer"] == "") {
+			if cfg.CertManagerIssuer != nil {
+				ingress.Annotations["cert-manager.io/cluster-issuer"] = *cfg.CertManagerIssuer
+			}
 		}
-		hostnameSeen[rule.Host] = struct{}{}
-		svc.Status.Endpoints = append(svc.Status.Endpoints, v1.Endpoint{
-			Address:         rule.Host,
-			PublishProtocol: proto,
-		})
-	}
 
-	result = append(result, &networkingv1.Ingress{
-		TypeMeta: metav1.TypeMeta{},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      svc.Name,
-			Namespace: svc.Namespace,
-			Labels:    svc.Spec.Labels,
-			Annotations: labels.Merge(svc.Spec.Annotations, map[string]string{
-				labels.AcornTargets: string(targetJSON),
-			}),
-		},
-		Spec: networkingv1.IngressSpec{
-			IngressClassName: ingressClassName,
-			Rules:            rules,
-			TLS:              ingressTLS,
-		},
-	})
+		result = append(result, ingress)
 
-	result = append(result, secrets...)
+		result = append(result, secrets...)
+	}
 
 	return
 }
 
-func setupCertManager(serviceName string, annotations map[string]string, rules []networkingv1.IngressRule, tls []networkingv1.IngressTLS) []networkingv1.IngressTLS {
-	if annotations["cert-manager.io/cluster-issuer"] == "" && annotations["cert-manager.io/issuer"] == "" {
+func setupCertManager(serviceName string, annotations map[string]string, rules []networkingv1.IngressRule, tls []networkingv1.IngressTLS, useCertManager bool) []networkingv1.IngressTLS {
```

Comment:
Feels weird passing in `useCertManager` to `setupCertManager`.. like if you don't want to use cert manager, then don't call this function :sweat_smile: 
It's fine though.

Reply:
Agreed

Reply:
Makes sense. I have moved out the condition to the toplevel.